src/app/settings/admin/rule-groups/component.ts

```diff
@@ -0,0 +1,186 @@
+import _ from 'lodash';
+import {combineLatest, Observable, Subject} from 'rxjs';
+import {filter, map, switchMap, take, takeUntil, tap} from 'rxjs/operators';
+import {AdminRuleGroupDialog, Mode} from './rule-group-dialog/component';
+
+@Component({
+  selector: 'km-admin-settings-rule-groups',
+  styleUrls: ['style.scss'],
+  templateUrl: 'template.html',
+})
+export class AdminSettingsRuleGroupsComponent implements OnInit, OnChanges, OnDestroy {
+  dataSource = new MatTableDataSource<AdminRuleGroup>();
+  displayedColumns: string[] = ['name', 'type', 'seed', 'actions'];
+  seeds: string[] = [];
+  seedFilter: string;
+  typeFilter: string;
+  settings: UserSettings;
+  ruleGroupTypes = Object.values(RuleGroupType);
+  adminRuleGroups: AdminRuleGroup[] = [];
+
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+  @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
+
+  private _unsubscribe = new Subject<void>();
+
+  constructor(
+    private readonly _matDialog: MatDialog,
+    private readonly _mlaService: MLAService,
+    private readonly _userService: UserService,
+    private readonly _notificationService: NotificationService,
+    private readonly _datacenterService: DatacenterService
+  ) {}
+
+  ngOnInit(): void {
+    this.dataSource.data = this.adminRuleGroups || [];
+    this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
+    this.sort.active = 'name';
+    this.sort.direction = 'asc';
+
+    this._userService.currentUserSettings.pipe(takeUntil(this._unsubscribe)).subscribe(settings => {
+      this.settings = settings;
+      this.paginator.pageSize = settings.itemsPerPage;
+      this.dataSource.paginator = this.paginator; // Force refresh.
+    });
+
+    this._datacenterService.seeds
+      .pipe(tap(seeds => (this.seeds = seeds)))
+      .pipe(
+        switchMap(seeds =>
+          combineLatest([
+            ...seeds.map<Observable<[string, RuleGroup[]]>>(seed =>
+              this._mlaService.adminRuleGroups(seed).pipe(map(ruleGroup => [seed, ruleGroup]))
+            ),
+          ])
+        )
+      )
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(ruleGroupTuples => {
+        this.adminRuleGroups = ruleGroupTuples
+          .map(tuple => tuple[1].map<[string, RuleGroup]>(ruleGroup => [tuple[0], ruleGroup]))
+          .flatMap(tuples =>
+            tuples.map(
+              tuple =>
+                ({seed: tuple[0], name: tuple[1].name, data: tuple[1].data, type: tuple[1].type} as AdminRuleGroup)
+            )
+          );
+
+        this.dataSource.data = this.adminRuleGroups;
+      });
+  }
+
+  ngOnChanges(): void {
+    this.filter();
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  isPaginatorVisible(): boolean {
+    return (
+      this.adminRuleGroups &&
+      this.adminRuleGroups.length > 0 &&
+      this.paginator &&
+      this.adminRuleGroups.length > this.paginator.pageSize
```

Comment:
`this.adminRuleGroups?.length > this.paginator?.pageSize` seems to cover it all assuming that page size is at least equal to 0.

Reply:
yep, sounds good to me 👍